climate.py

```diff
@@ -148,17 +155,34 @@ def preset_mode(self):
     def preset_modes(self):
         """Return the preset modes, comfort, away etc"""
         return PRESET_MODES
+        
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self._current_temperature is not None:
+            return float(self._current_temperature)
+        return None
 
     def set_hvac_mode(self, hvac_mode):
-        """Noop for the time being"""
-        hvac_mode = hvac_mode.lower()
+        if hvac_mode == HVAC_MODE_AUTO:
+            self.set_preset_mode('')
```

Comment:
```suggestion
            self.set_preset_mode(PRESET_NONE)
```
Suggest using PRESET_NONE instead of nothing